src/main/java/com/isroot/stash/plugin/YaccHook.java

```diff
@@ -66,7 +66,16 @@ private RepositoryHookResult handleRepositoryPush(
             @Nonnull RepositoryPushHookRequest repositoryPushHookRequest) {
         final Settings settings = context.getSettings();
 
-        return yaccService.check(context, repositoryPushHookRequest, settings);
+        RepositoryHookResult result;
+        try {
+            result = yaccService.check(context, repositoryPushHookRequest, settings);
+        } catch (TimeLimitedMatcherFactory.RegExpTimeoutException e) {
```

Comment:
Ideally we should have a unit test for this to ensure `TimeLimitedMatcherFactory` is working as intended and the exception is being caught with a message being shown to the committer.

[Similar to other question)[https://github.com/sford/yet-another-commit-checker/pull/223#issuecomment-532205767)... is there an example regex and commit message that can trigger regex dos reliably for use for a unit test?

Reply:
Currently the other commit regex tests are located at in [YaccServiceImplTest](https://github.com/sford/yet-another-commit-checker/blob/75c5cd0c65fafa1bbcc7b410408dbb44915c24fa/src/test/java/ut/com/isroot/stash/plugin/YaccServiceImplTest.java#L321-L346)

Reply:
Hi @sford , I'll take a look and include a unit test. In our case, we didn't find what regex(s) were causing the issues. We only found that there were many threads causing high CPU usage. Our first approach was adding some additional logs to measure the time taken to evaluate the regex, but as we were already using our custom plugin, we went directly for the fix.

In my tests, using:
`^(a+)+$ `
as pattern, and a commit message like:
`aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaX`

would take more than 2 seconds and throw the exception after making the code changes. Testing in https://regex101.com/ that regex with `aaaaaaaaaaaaaaaaaX` as input will report 393216 steps and 223 ms, and adding additional "a" to the input string will be reported as "catastrophic backtrack". Apparently every additional "a" in the input will double the number of steps.

Reply:
Added a unit test for this example regular expression. One thing to note: I was using JDK 12 in my latest builds instead of JDK 8 and I was not able to make the test fail! I just noticed that using JDK 8 the problem can be reproduced, but it has been fixed in JDK 12 (or maybe an earlier version). So that is good news! We should keep the test for now, but maybe the fix is not needed anymore when using a newer Java version, or at least for this specific regex (but others might also cause catastrophic backtrack, I didn't find anything relating to this issue being fixed in newer Java versions)

Reply:
According to https://stackoverflow.com/questions/53048859/is-java-redos-vulnerable, Java 11 is not vulnerable to this regex, but changing it to `((a+)+)+` raises the issue again. So I am using a more complex expression to make the test work on newer JDK versions.